Dockerfile

```diff
@@ -1,10 +1,12 @@
 FROM node:14.15.3
 
-RUN mkdir -p /opt/notes-app
 WORKDIR /opt/notes-app
 
 COPY package.json package-lock.json ./
 
 RUN npm install
 
-CMD ["npm", "run", "start"]
+COPY . .
+
+ENTRYPOINT [ "npm", "run" ]
+CMD [ "start:prod" ]
```

Comment:
Could we leave this to start in dev and parameterize this instead? Eg using `NODE_ENV`

Reply:
Well my thought was that if in the future scenario where this were published to a docker hub or something and you were to just run it in a prod-like scenario, your compose could look like this (very minimal):

```
app:
  image: react-server-components
  env_file: .env
  ports:
	- <external>:4000
```

However you'll note in _our_ docker-compose, it overrides the command to run `start` (dev workflow).

I'm open to changes though! Can you elaborate on how you'd want to switch between workflows? based solely on `NODE_ENV`?

Otherwise the simple thing to do is just revert back to `start` in the dockerfile, then in docker-compose the `command` becomes unnecessary, the example above (prod workflow) you would add `command: [start:prod]`.